src/app.py

```diff
@@ -2311,7 +2311,7 @@ def validate_donors(headers, records):
 # Determines if a dataset is Primary. If the list returned from the neo4j query is empty, the dataset is not primary
 def dataset_is_primary(dataset_uuid):
     with neo4j_driver_instance.session() as neo_session:
-        q = (f"MATCH (ds:Dataset {{uuid: '{dataset_uuid}'}})<-[:ACTIVITY_OUTPUT]-(:Activity)<-[:ACTIVITY_INPUT]-(s:Sample) RETURN ds.uuid")
+        q = (f"MATCH (ds:Dataset {{uuid: '{dataset_uuid}'}})<-[:ACTIVITY_OUTPUT]-(a:Activity) WHERE NOT toLower(a.creation_action) ENDS WITH 'process' RETURN ds.uuid")
```

Comment:
@shirey from your early comment on Slack:
>The result of a “Multi-Assay Split” dataset isn’t a derived/processed dataset so should not be included in this task.

Does this mean the two component datasets generated by a "Multi-Assay Split" Activity are NOT considered as primary datasets?

I had the same question in entity-api PR due to the same use of negation clause: https://github.com/hubmapconsortium/entity-api/pull/559#discussion_r1374900492

Reply:
@yuanzhou “Multi-Assay Split” datasets are primary datasets.